### PR TITLE
Add a review guide to docs

### PR DESCRIPTION
**Pull request by DonGiovanni83**
In order to streamline reviews a short guide for reviewing developers has been added.

### CODE REVIEW

File: docs/docsite/rst/development_guide.rst [hunk 1]
```diff
@@ -128,3 +128,34 @@ directory structure like this:
...
+
+
+Reviewing Code
+=================
+
+Prerequisite for a successful review is to have setup your environment according
+to the section *Local Setup*. To review changes of other contributors use these
+steps as a review guideline:
+
+1. Check out the new branch
```
Comment (nerrehmit) [inline docs/docsite/rst/development_guide.rst:140]: Wouldn't the reviewer of the PR have to first check out the forked repository of the person that created the PR.
If that is the case is there any way we can simplify this?

Or is the workflow really:
- clone the forked repo
- switch to the desired branch
- run tests/ build-doc, etc?

**Review by nerrehmit — COMMENTED** [review]
Sounds good in general 👍 We might consider adding a section about the tests that are already present in the repo.

When we start adding more code/ modules we can always expand upon the necessary checks one has to do before merging.

File: docs/docsite/rst/development_guide.rst (hunk 1 above)
Comment (DonGiovanni83) [reply docs/docsite/rst/development_guide.rst]: Thanks for the hint. I added the missing fork checkout to the first step.

**Review by nerrehmit — APPROVED** [review]
LGTM! Thanks for the small changes.

**Pull request merged**